source/common/http/http2/metadata_encoder.cc

```diff
@@ -0,0 +1,76 @@
+#include "common/http/http2/metadata_encoder.h"
+
+#include "common/common/assert.h"
+
+#include "nghttp2/nghttp2.h"
+
+namespace Envoy {
+namespace Http {
+namespace Http2 {
+
+bool MetadataEncoder::createPayload(MetadataMap& metadata_map) {
+  if (payload_.length() > 0 || metadata_map.empty()) {
+    // |payload_| is not empty or |metadata_map| is empty. Return failure.
+    ENVOY_LOG(error, "payload_ size is {}, metadata_map is {}empty", payload_.length(),
+              metadata_map.empty() ? "" : "not ");
+    return false;
+  }
+
+  bool success = createHeaderBlockUsingNghttp2(metadata_map);
+  if (!success || !hasNextFrame()) {
```

Comment:
ASSERT? Should this ever happen?

Reply:
I think the call can fail when metadata exceeds the size limit. But I removed the hasNextFrame() check because it is redundant.